Automattic/jetpack
# Update/newsletter category settings

Fixes https://github.com/Automattic/loop/issues/281

## Proposed changes:

* Change the newsletter categories toggle to save only when "Save Settings" is clicked.
* Add "Which categories would you like to allow people to subscribe to? Select all that apply:" above the newsletter category selection.
* Show an error message when newsletter categories are enabled, but none are selected: "Please select at least one category when newsletter categories are enabled."
* No go: adding testing for newsletter settings. I attempted to but mocking all the required data got complicated fast. There are tests for settings in general.
* Note there's only one Save Settings button you can use to save the categories settings.

<img width="1375" alt="Screen Shot 2024-12-23 at 5 47 59 PM" src="https://github.com/user-attachments/assets/022d7f27-9bb3-4584-b9b2-861cc8f8c6cd" />

### Other information:


## Jetpack product discussion



p1734114241935889/1733936868.332319-slack-C083ZPVVDTK

## Does this pull request change what data or activity we track or use?



N/A

## Testing instructions:






- On a Jetpack or Atomic site, go to /wp-admin/admin.php?page=jetpack#/newsletter and scroll to Newsletter categories.
- Enable categories.
- Without selecting any categories, click "Save settings."
- You should see an error, and settings won't save.
- Select one or more categories.
- Settings should save successfully.
- Disable categories.
- Settings should save successfully.
- Start with a site that currently has categories enabled but none selected. Check that the page still loads and you can select categories.
